### PR TITLE
systemd generator: force run container detached if CreateCommand has no detach param

### PR DESCRIPTION
**Pull request by ttys3**
since we use systemd service `Type=forking`
when we generated systemd service file using the `--new` param like
```
 sudo podman generate systemd --name --new ctrName
```
we got `ExecStart=/usr/bin/podman run --conmon-pidfile ...`
if somehow the container's `CreateCommand` has no `-d` or `--detach` param,
podman will run the container in default attached mode.
so systemd will wait the podman command exit until failed with timeout error.

quick reproduce the problem:
```
sudo podman pull nginxdemos/hello
sudo podman create --name ngxdemo -p 8086:80 nginxdemos/hello
cd /etc/systemd/system/
sudo podman generate systemd --name --new --restart-policy on-failure -t 3 --files ngxdemo
sudo podman rm ngxdemo

sudo systemctl start container-ngxdemo.service
Job for container-ngxdemo.service failed because a timeout was exceeded.
See "systemctl status container-ngxdemo.service" and "journalctl -xe" for details.
```

how to avoid the problem:

solution 1: merge this PR -_- and everything goes fine

solution 2: warn user to remember add `-d` when create the container
and user need to do is just add the `-d` param to the `create` command:
```
sudo podman pull nginxdemos/hello
sudo podman create -d --name ngxdemo -p 8086:80 nginxdemos/hello
cd /etc/systemd/system/
sudo podman generate systemd --name --new --restart-policy on-failure -t 3 --files ngxdemo
sudo podman rm ngxdemo
sudo systemctl start container-ngxdemo.service
# everything goes fine
```

### CODE REVIEW

File: pkg/systemd/generate/systemdgen.go [hunk 1]
```diff
@@ -164,6 +164,17 @@ func CreateContainerSystemdUnit(info *ContainerInfo, opts Options) (string, erro
 			"--cidfile", "%t/%n-cid",
 			"--cgroups=no-conmon",
 		}
+
+		// Check detach param
```
Comment (vrothberg) [inline pkg/systemd/generate/systemdgen.go]: Please add a more verbose comment why we want to force `--detach`.

**Comment by rhatdan** [comment]
Thanks @ttys3 Interesting change.

**Comment by mheon** [comment]
LGTM once comments from @vrothberg are addressed

File: pkg/systemd/generate/systemdgen.go (hunk 1 above)
Comment (ttys3) [reply pkg/systemd/generate/systemdgen.go]: thanks for the guide. I've forced pushed the PR

**Comment by vrothberg** [comment]
e2e tests are not yet happy:

> [+1126s] [Fail] Podman generate systemd [It] podman generate systemd --new with explicit detaching param in middle 
[+1126s] /var/tmp/go/src/github.com/containers/libpod/test/e2e/generate_systemd_test.go:223

**Comment by ttys3** [comment]
> e2e tests are not yet happy:
> 
> > [+1126s] [Fail] Podman generate systemd [It] podman generate systemd --new with explicit detaching param in middle
> > [+1126s] /var/tmp/go/src/github.com/containers/libpod/test/e2e/generate_systemd_test.go:223

fixed. 
but I see another two tests failed (seems not related to my PR):

special_testing_rootless name:remote Failing after 21m — Task Summary
test fedora-30 name:remote Failing after 25m — Task Summary

**Comment by mheon** [comment]
F30 remote is a known flake.

Rootless might be a real but. Restarted to see if it was a flake, but wouldn't be surprised if it reoccurs.

**Comment by vrothberg** [comment]
Another restart :)

**Comment by edsantiago** [comment]
This has broken CI. Is anyone working on a fix?

**Comment by vrothberg** [comment]
> This has broken CI. Is anyone working on a fix?

Entirely my bad as we should have rebased this PR before merging. https://github.com/containers/libpod/pull/5514 is fixing CI.

**Pull request merged**